4_evaluate/src/interpolate_predictions.py

```diff
@@ -0,0 +1,225 @@
+import numpy as np
+import pandas as pd
+from tqdm import tqdm
+
+"""
+Evaluate predictions the same way as lake-temperature-process-models:
+Interpolate ML output to observation depths.
+"""
+
+def interpolate_predictions(predictions_filepath,
+                            sequence_indices,
+                            date_indices,
+                            depth_above_indices,
+                            depth_below_indices,
+                            above_factors,
+                            below_factors,
+                            use_obs):
+    """
+    Load and interpolate predictions from predicted depths to observation depths
+
+    :param predictions_filepath: Path to saved unnormalized predictions
+    :param sequence_indices: Index of every observation's corresponding sequence in predictions array
+    :param date_indices: Index for every observation within its sequence
+    :param depth_above_indices: Indexing array for the predictions above the observation locations
+    :param depth_below_indices: Indexing array for the predictions below the observation locations
+    :param above_factors: Interpolation factors for the predictions above the observations
+    :param below_factors: Interpolation factors for the predictions below the observations
+    :param use_obs: Indexing array for observation locations to interpolate to
+    :returns: Array of temperature predictions interpolated to observation depths
+
+    """
+    # Load predictions
+    predictions_npz = np.load(predictions_filepath)
+    ta = predictions_npz['predictions'][sequence_indices[use_obs], date_indices[use_obs], depth_above_indices[use_obs]]
+    tb = predictions_npz['predictions'][sequence_indices[use_obs], date_indices[use_obs], depth_below_indices[use_obs]]
+    # Interpolate
+    predictions = ta*above_factors[use_obs] + tb*below_factors[use_obs]
+    return predictions
+
+
+def get_interpolated_predictions(predictions_filepath,
+                                 metadata_filepath,
+                                 dataset_filepath,
+                                 obs_file,
+                                 interpolated_predictions_filepath,
+                                 batch_size=5000):
+    """
+    Interpolate ML output to observation depths.
+
+    :param predictions_filepath: Path to unnormalized predictions for one
+        dataset (train, validation, or test)
+    :param metadata_filepath: Path to model training metadata npz for one dataset
+    :param dataset_filepath: Path to input for one dataset
```

Comment:
"path to input for one dataset" is vague/not as informative as the other lines of documentation

Reply:
Agreed - reworded